Ginger/GingerCore/Environments/Database.cs

```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
                 //DbTypes = value;
             }
         }
-
+        public bool CheckUserCredentialsInTNS()
+        {
+            string tnsCalculated = GetDecryptOrCalcluatedValue(TNSCalculated);
+            if (!string.IsNullOrEmpty(tnsCalculated) && tnsCalculated.ToLower().Contains("data source=") && tnsCalculated.ToLower().Contains("password=") && tnsCalculated.ToLower().Contains("user id="))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+        public string GetDecryptOrCalcluatedValue(string calculatedValue)
+        {
+            bool res = false;
+            String deCryptValue = EncryptionHandler.DecryptString(calculatedValue, ref res, false);
+            if (res == true)
+            {
+                return deCryptValue;
+            }
+            else
+            {
+                return calculatedValue;
+            }
+        }
         public string NameBeforeEdit;
 
        
         public string GetConnectionString()
         {
             string connStr = null;
-            bool res;
-            res = false;
+            
+            //Decryption
+            string User = GetDecryptOrCalcluatedValue(UserCalculated);
+            string Pass = GetDecryptOrCalcluatedValue(PassCalculated);
+            string TNS = GetDecryptOrCalcluatedValue(TNSCalculated);
+            string ConnString = GetDecryptOrCalcluatedValue(ConnectionStringCalculated);
 
-            if (String.IsNullOrEmpty(ConnectionStringCalculated) == false)
+            if (String.IsNullOrEmpty(ConnString) == false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)